lib/logstash/filters/csv.rb

```diff
@@ -70,84 +91,56 @@ def register
         raise LogStash::ConfigurationError, "#{type} is not a valid conversion type."
       end
     end
-  end # def register
+
+    # @convert_symbols contains the symbolized types to avoid symbol conversion in the transform method
+    @convert_symbols = @convert.inject({}){|result, (k, v)| result[k] = v.to_sym; result}
+
```

Comment:
I all of a tizzy about the CONVERTERS, VALID_CONVERT_TYPES, @convert_symbols etc.

On the one hand I don't want to generate more work but on the other I'm Gandalf - "Thou shall not pass" :smile:

My Gandalf proposal
```ruby
  # List of valid conversion types used for the convert option
  # delete me VALID_CONVERT_TYPES = [ "integer", "float", "date", "date_time", "boolean" ].freeze

  CONVERTERS = {
    "integer" => lambda do |value|
      CSV::Converters[:integer].call(value)
    end,

    "float" => lambda do |value|
      CSV::Converters[:float].call(value)
    end,

    "date" => lambda do |value|
      CSV::Converters[:date].call(value)
    end,

    "date_time" => lambda do |value|
      CSV::Converters[:date_time].call(value)
    end,

    "boolean" => lambda do |value|
       value = value.strip.downcase
       return false if value == "false"
       return true  if value == "true"
       return value
    end
  }
  CONVERTERS.default(lambda { |v| v })
  CONVERTERS.freeze
  
  def register
    # validate conversion types to be the valid ones.

    bad_types = @convert.values.select do |column, type|
      !CONVERTERS.key?(type)
    end

    unless bad_types.empty?
      raise(LogStash::ConfigurationError,
        "These types are not valid conversion types: #{bad_types.join(', "')}")
    end

    # @convert_symbols contains the symbolized types to avoid symbol conversion in the transform method
    # delete me @convert_symbols = @convert.inject({}){|result, (k, v)| result[k] = v.to_sym; result}

    # make sure @target is in the format [field name] if defined, i.e. surrounded by brakets
    @target = "[#{@target}]" if @target && @target !~ /^\[[^\[\]]+\]$/
  end

  def transform(field_name, value)
    CONVERTERS[@convert[field_name]].call(value)
  end
```
feel free to punt this to a new issue though.